lib/ensure/link.js

```diff
@@ -14,9 +14,9 @@ function createLink (srcpath, dstpath, callback) {
     })
   }
 
-  pathExists(dstpath, (err, destinationExists) => {
-    if (err) return callback(err)
-    if (destinationExists) return callback(null)
+  fs.lstat(dstpath, (err, stats) => {
+    if (!err && stats.isSymbolicLink()) return callback(null)
```

Comment:
This doesn't work for links (hard links) since `stats.isSymbolicLink()` would return `false` for hard links as symlinks and hard links are different types. The thing is, there is no easy way to check if a file is a hard link! One way to do that would be by checking the entire file system :exploding_head:  to see if there is any file that has the same inode as the hard link since hard link and the original file have the same inodes on the file system.

I am personally not sure what would be the best approach here.

1. We could check if dest exists and if is symlink, then throw error.
2. Or we allow the operation proceeds which means if dest exists, a `file already exists` error will be thrown.

Let me know what you guys think!

Reply:
Ah, yeah, unit tests should be modified to cover this case, I didn't think this through.

Actually, I'm specifically wondering if we shouldn't be checking that it's a hard link between given src and dest, since we're trying to ensure it. Same actually goes for symlinks.